plugins/fabric8-end2end-flow/src/main/java/com/redhat/che/end2end/End2EndFlowService.java

```diff
@@ -0,0 +1,217 @@
+    this.reCaptchaSiteKey = siteKey;
+    this.reCaptchaSecretKey = secretKey;
+    this.staticFilesFilters = new HashMap<>();
+    String siteKeyDecl;
+    if (siteKey != null && !siteKey.isEmpty()) {
+      siteKeyDecl = "var siteKey='" + siteKey + "';";
+    } else {
+      siteKeyDecl = "var siteKey='';";
+      LOG.warn("No ReCaptcha site key was provided. ReCaptcha user verification is disabled !");
+    }
+    staticFilesFilters.put(
+        "files/provision.html", line -> line.replace("const siteKey;", siteKeyDecl));
+  }
+
+  @GET
+  @Path("site-key")
+  public String siteKey() {
+    return reCaptchaSiteKey == null ? "" : reCaptchaSiteKey;
+  }
+
+  @POST
+  @Path("verify")
+  @Consumes("text/plain")
+  @Produces(APPLICATION_JSON)
+  public Response verifyReCaptchaToken(String token) {
+    int responseCode = 500;
+    try {
+      URL obj = new URL("https://www.google.com/recaptcha/api/siteverify");
+      HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
+
+      con.setRequestMethod("POST");
+
+      String urlParameters = "secret=" + reCaptchaSecretKey + "&response=" + token;
+
+      con.setDoOutput(true);
+      try (DataOutputStream wr = new DataOutputStream(con.getOutputStream())) {
+        wr.writeBytes(urlParameters);
+        wr.flush();
+      }
+
+      responseCode = con.getResponseCode();
+
+      if (responseCode == 200) {
+        StringBuffer response = new StringBuffer();
+        try (BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
+          String inputLine;
+          while ((inputLine = in.readLine()) != null) {
+            response.append(inputLine);
+          }
+        }
+        return Response.ok(response.toString()).build();
+      } else {
+        LOG.error(
+            "reCaptcha verification failed with the fllowing response code: {}", responseCode);
+      }
+    } catch (IOException e) {
+      LOG.error("Captcha verification failed with an exception", e);
+    }
+
+    return Response.status(responseCode).build();
+  }
+
+  private String getLog(String message, HttpServletRequest servletRequest) {
+    String ip = servletRequest.getHeader("X-Forwarded-For");
+    if (ip == null) {
+      ip = servletRequest.getRemoteAddr();
+    }
+
+    return new StringBuffer("[E2E Registration Flow - IP = ")
+        .append(ip == null ? "unknown" : ip)
+        .append("] ")
+        .append(message)
+        .toString();
+  }
+
+  @POST
+  @Path("warning")
+  @Consumes("text/plain")
+  public void warning(String message, @Context HttpServletRequest servletRequest) {
+    LOG.warn(getLog(message, servletRequest));
+  }
+
+  @POST
+  @Path("error")
+  @Consumes("text/plain")
+  public void error(String message, @Context HttpServletRequest servletRequest) {
+    LOG.error(getLog(message, servletRequest));
+  }
+
+  @GET
+  @Path("{path: files\\/.*}")
+  public Response staticResources(@PathParam("path") final String path) {
```

Comment:
It is not very common to host files using a REST service. Why not use servlet for that? I suppose there is ready servlet somewhere in Che for this purpose.

Reply:
Is it a problem ? It seems much easier to configure / maintain than using a Servlet, no ?

Additionally this was done this way in a similar context, and merged on upstream: https://github.com/eclipse/che/blob/07263f1e30089689d71b057f747a44a29283e3c4/multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakConfigurationService.java#L54

So if you don't mind, I'd rather let it like that.

Reply:
I wouldn't block the PR because of that, but I do consider binding of a component with a couple of lines easier to support than 60 lines of code.

Reply:
I can change it if you point me to the right Che ready-to-use file-serving Servlet component.
However let me note that the content of the files is optionally filtered before returning it. Would the out-of-the-box Che Servlet component provide this ability ?

Reply:
I'm OK to leave it. Don't see much sense to argue about it.